flink-runtime/src/test/java/org/apache/flink/runtime/messages/webmonitor/JobDetailsTest.java

```diff
@@ -103,4 +105,33 @@ public void testJobDetailsCompatibleUnmarshalling() throws IOException {
 
         assertEquals(expected, unmarshalled);
     }
+
+    @Test
+    public void testJobDetailsWithExecutionAttemptsMarshalling() throws JsonProcessingException {
+        Map<String, Map<Integer, Integer>> currentExecutionAttempts = new HashMap<>();
+        currentExecutionAttempts.computeIfAbsent("a", k -> new HashMap<>()).put(1, 2);
+        currentExecutionAttempts.computeIfAbsent("a", k -> new HashMap<>()).put(2, 4);
+        currentExecutionAttempts.computeIfAbsent("b", k -> new HashMap<>()).put(3, 1);
+
+        final JobDetails expected =
+                new JobDetails(
+                        new JobID(),
+                        "foobar",
+                        1L,
+                        10L,
+                        9L,
+                        JobStatus.RUNNING,
+                        8L,
+                        new int[] {1, 3, 3, 4, 7, 4, 2, 7, 3, 3},
+                        42,
+                        currentExecutionAttempts);
+
+        final ObjectMapper objectMapper = RestMapperUtils.getStrictObjectMapper();
+
+        final JsonNode marshalled = objectMapper.valueToTree(expected);
+
+        final JobDetails unmarshalled = objectMapper.treeToValue(marshalled, JobDetails.class);
+
+        assertEquals(expected, unmarshalled);
```

Comment:
Flink is migrating to JUnit 5 so that all the new tests should use `JUnit5` and `assertJ`.
You can migrate the existing tests to JUnit5 in a  hotfix commit before making functional changes.